### PR TITLE
Dynatrace Output Plugin: Fixed behaviour when state map is cleared

### PR DESCRIPTION
**Pull request by thschue**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by ssoroka** [comment]
can you try merging the latest master into this branch?

File: plugins/outputs/dynatrace/dynatrace.go [hunk 1]
```diff
@@ -195,17 +195,26 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
...
-						// only send a counter if a lastvalue is found in the map
-						// if last value is found we can calc and send the delta value
-						if v, err := strconv.ParseFloat(lastvalue, 32); err == nil {
-							if v2, err := strconv.ParseFloat(value, 32); err == nil {
-								fmt.Fprintf(&buf, "%s%s count,delta=%f\n", metricID, tagb.String(), v2-v)
-							}
+						// Convert Strings to Floats
+						floatLastValue, err := strconv.ParseFloat(lastvalue, 32)
+						if err != nil {
+							d.Log.Debugf("Could not parse last value: %s", lastvalue)
```
Comment (ssoroka) [inline plugins/outputs/dynatrace/dynatrace.go:205]: allowing it to continue will leave floatLastValue at 0. is this ok?

File: plugins/outputs/dynatrace/dynatrace.go [hunk 2]
```diff
@@ -195,17 +195,26 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
...
+						}
+						floatCurrentValue, err := strconv.ParseFloat(value, 32)
+						if err != nil {
+							d.Log.Debugf("Could not parse current value: %s", value)
```
Comment (ssoroka) [inline plugins/outputs/dynatrace/dynatrace.go:209]: allowing it to continue will leave floatCurrentValue at 0. is this ok?

File: plugins/outputs/dynatrace/dynatrace.go [hunk 3]
```diff
@@ -195,17 +195,26 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
...
+						}
+						if floatCurrentValue > floatLastValue {
```
Comment (ssoroka) [inline plugins/outputs/dynatrace/dynatrace.go:211]: You never print a negative delta? abs() better?

Comment (thschue) [reply plugins/outputs/dynatrace/dynatrace.go:211]: This behaviour is intended at the moment ...

File: plugins/outputs/dynatrace/dynatrace.go (hunk 2 above)
Comment (thschue) [reply plugins/outputs/dynatrace/dynatrace.go:209]: This behaviour is intended at the moment ...

File: plugins/outputs/dynatrace/dynatrace.go (hunk 1 above)
Comment (thschue) [reply plugins/outputs/dynatrace/dynatrace.go:205]: This behaviour is intended at the moment ...

**Review by ssoroka — APPROVED** [review]

**Pull request merged**